src/angular/components/send/add-edit.component.ts

```diff
@@ -156,24 +156,28 @@ export class AddEditComponent implements OnInit {
         // Parse dates
         if (!this.isDateTimeLocalSupported) {
             const deletionDateParts = this.dateToSplitString(this.send.deletionDate);
-            this.deletionDateFallback = deletionDateParts[0];
-            this.deletionTimeFallback = deletionDateParts[1];
+            if (deletionDateParts !== undefined && deletionDateParts.length > 0) {
+                this.deletionDateFallback = deletionDateParts[0];
+                this.deletionTimeFallback = deletionDateParts[1];
+            }
+
             const expirationDateParts = this.dateToSplitString(this.send.expirationDate);
-            this.expirationDateFallback = expirationDateParts[0];
-            this.expirationTimeFallback = expirationDateParts[1];
+            if (expirationDateParts !== undefined && expirationDateParts.length > 0) {
+                this.expirationDateFallback = expirationDateParts[0];
+                this.expirationTimeFallback = expirationDateParts[1];
+            }
         } else {
             this.deletionDate = this.dateToString(this.send.deletionDate);
             this.expirationDate = this.dateToString(this.send.expirationDate);
         }
     }
 
     async submit(): Promise<boolean> {
-        if (!this.isDateTimeLocalSupported && this.expirationDateTimeFallback !== null) {
-            this.expirationDate = this.expirationDateTimeFallback;
-        }
-
-        if (!this.isDateTimeLocalSupported && this.deletionDateTimeFallback !== null) {
+        if (!this.isDateTimeLocalSupported) {
             this.deletionDate = this.deletionDateTimeFallback;
+            if ((this.editMode && this.expirationDateFallback != null) || this.expirationDateSelect === 0) {
+                this.expirationDate = this.expirationDateTimeFallback;
```

Comment:
`this.expirationDateTimeFallback` return `'T'` when you clear out the expiration parts.

Probably the parts need to be linked so that clearing one clears the other. Also there is a clear link which does not work right now.

I'm seeing some stuff to get rid of the clear `x` entirely for date and time inputs, that might be the way to go